experimental/typed-dsl/Compo.idr

```diff
@@ -0,0 +1,33 @@
+module Compo
+
+import Service
+import Numeric
+import Halfer
+import Twicer
+import Incrementer
+
+%default total
+
+
+
+-- composition of Twicer and Incrementer
+compo1 : Integer -> Service (Integer)
+compo1 a = do
+            n <- twicerService a
+            incrementerService n
+
+-- composition of Twicer, Halfer, and Incrementer
+compo2 : Integer -> Service (Integer)
+compo2 a = do
+            i <- twicerService a
+            j <- Promise (MkContract ?check1)
+            k <- halferService j
+            incrementerService k
+
+
+-- invalid composition of Twicer, Incrementer, and Halfer
+compo3 : Integer -> Service (Integer)
+compo3 a = do
+            n <- twicerService a
+            m <- incrementerService (cast $ fst n)
```

Comment:
Why `fst` is used?

When compiling (in idris2) I get the following error

```
6/6: Building Compo (Compo.idr)
Error: While processing right hand side of compo3. Sorry, I can't find any elaboration which works. All errors:
If Builtin.fst: When unifying Integer and (?a, ?b).
Mismatch between: Integer and (?a, ?b).

Compo.idr:32:49--32:50
    |
 32 |             m <- incrementerService (cast $ fst n)
    |                                                 ^

If Builtin.DPair.DPair.fst: When unifying Integer and DPair ?a ?p.
Mismatch between: Integer and DPair ?a ?p.

Compo.idr:32:49--32:50
    |
 32 |             m <- incrementerService (cast $ fst n)
    |                                                 ^
```

Reply:
@sroberti, do you think you could push a fix soon? I'm on my way to try to integrate it to my registry example.